core/config/project_settings.cpp

```diff
@@ -508,6 +508,21 @@ void ProjectSettings::_convert_to_last_version(int p_from_version) {
 			}
 		}
 	}
+	if (p_from_version <= 5) {
```

Comment:
You need to increment project version, otherwise this code will run every time.

Reply:
I didn't want to unnecessarily increment the project version number.

This conversion can be implemented without incrementing the project version.
Before this PR, possible values of device id in the project file are: -1, 0, ..., 7
With this PR, possible values of device id in the project file are: -3, 0, ..., 7
So the conversion doesn't affect projects, that are saved with this PR. The conversion just makes sure, that all projects use -3 for "all devices" after they are loaded.

Also it is compatible with future project version increments.

The conversion happens only once during loading, so there shouldn't be any noticeable delay.

Reply:
At least wrap it in DISABLE_DEPRECATED check then.

Reply:
Should I wrap the section above also in a `DISABLE_DEPRECATED` check?

https://github.com/godotengine/godot/blob/44fa552343722bb048e2d7c6d3661174a95a8a3c/core/config/project_settings.cpp#L497-L509

Reply:
tbh the other sections is irrelevant at this point. Version 5 is used since 4.0.
But yeah, it should be wrapped too.